internal/provider/utils_wait.go

```diff
@@ -754,9 +754,9 @@ func waitForFlinkStatementToBeUpdated(ctx context.Context, c *FlinkRestClient, s
 		targetStates = []string{stateStopped}
 		targetStatusMessage = stateStopped
 	} else {
-		pendingStates = []string{stateStopped, stateResuming}
-		targetStates = []string{statePending, stateRunning, stateCompleted}
-		targetStatusMessage = fmt.Sprintf("%s, %s or %s", statePending, stateRunning, stateCompleted)
+		pendingStates = []string{statePending, stateStopped, stateFailed}
```

Comment:
`stateFailed` seems kind of unintuitive for a pending state. Can a failed statement transition to a running or completed statement?

Reply:
Yes, what you mention is exactly matching the design.

In the case of an service account (without proper authorization level) assigned to a stopped statement to resume, the statement will transit to `stateFailed`, then we can resume such `stateFailed` by updating the service account.

Reply:
+1, AFAIK `pendingStates` is a list of states where TF should continue sending `GET` requests, but it seems like for `stateFailed` we should fail early.

Reply:
Good catch, `stateFailed` shouldn't be part of the `pendingStates` for sure.

On the other hand, I feel `stateFailed` shouldn't be part of the `targetStates` either. Otherwise it will trigger ambiguity since we have this logic:

```
if statement.Status.GetPhase() == statePending || statement.Status.GetPhase() == stateRunning || statement.Status.GetPhase() == stateCompleted || statement.Status.GetPhase() == stateStopped || statement.Status.GetPhase() == stateStopping {
  return statement, statement.Status.GetPhase(), nil
} else if statement.Status.GetPhase() == stateFailed || statement.Status.GetPhase() == stateFailing {
  return nil, stateFailed, fmt.Errorf("flink Statement %q provisioning status is %q", statementName, statement.Status.GetPhase())
}
```
I'll simply remove `stateFailed` from both, and rely on the error out to remind user about an undesired status change.